File: app/src/main/java/com/karntrehan/posts/base/di/AppModule.java
package com.karntrehan.posts.base.di;

import android.content.Context;

import com.karntrehan.posts.PostApp;

import dagger.Module;
import dagger.Provides;

/**
 * Created by karn on 03-06-2017.
 */
@Module
public class AppModule {

    private final PostApp postApp;

    public AppModule(PostApp postApp) {
        this.postApp = postApp;
    }

    @Provides
    @AppScope
    Context application() {
        return postApp.getApplicationContext();
    }
}

File: core/src/main/java/com/karntrehan/posts/core/di/AppModule.kt
package com.karntrehan.posts.core.di

import android.content.Context
import com.karntrehan.posts.core.networking.AppScheduler
import com.karntrehan.posts.core.networking.Scheduler
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule(val context: Context) {
    @Provides
    @Singleton
    fun providesContext(): Context {
        return context
    }

    @Provides
    @Singleton
    fun scheduler(): Scheduler {
        return AppScheduler()
    }
}